dotnet/src/dotnetcore/GxNetCoreStartup/SessionHelper.cs

```diff
@@ -48,6 +41,80 @@ private IDistributedCache GetTenantCache()
 		public Task SetAsync(string key, byte[] value, DistributedCacheEntryOptions options, CancellationToken token = default) => GetTenantCache().SetAsync(key, value, options, token);
 	}
 
+	public class CustomRedisSessionStore : IDistributedCache
+	{
+		private readonly IDatabase _db;
+		private readonly TimeSpan _idleTimeout;
+		private readonly TimeSpan _refreshThreshold;
+		private readonly string _instanceName;
+
+		public CustomRedisSessionStore(string connectionString, TimeSpan idleTimeout, string instanceName)
+		{
+			var mux = ConnectionMultiplexer.Connect(connectionString);
+			_db = mux.GetDatabase();
+			_idleTimeout = idleTimeout;
+			_refreshThreshold = TimeSpan.FromTicks((long)(idleTimeout.Ticks * 0.2));
+			_instanceName = instanceName ?? string.Empty;
+		}
+
+		private string FormatKey(string key) => string.IsNullOrEmpty(_instanceName) ? key : $"{_instanceName}:{key}";
+
+		public byte[] Get(string key) => _db.StringGet(FormatKey(key));
+
+		public async Task<byte[]> GetAsync(string key, CancellationToken token = default)
+		{
+			string redisKey = FormatKey(key);
+			var value = await _db.StringGetAsync(redisKey);
+
+			await RefreshKeyIfNeededAsync(redisKey);
+
+			return value;
+		}
+
+		public void Refresh(string key)
+		{
+			string redisKey = FormatKey(key);
+
+			var ttl = _db.KeyTimeToLive(redisKey);
+
+			if (ShouldRefreshKey(ttl))
+			{
+				_db.KeyExpire(redisKey, _idleTimeout);
+			}
```

Comment:
Same here. How checking for TTL before invoking expire results in performance gains, if both must reach a Redis?